src/engine/controls/clockcontrol.cpp

```diff
@@ -30,27 +43,145 @@ void ClockControl::trackLoaded(TrackPointer pNewTrack) {
+            if ((currentSample > prevIndicatorSamples) &&
+                    (currentSample <
+                            prevIndicatorSamples + m_blinkIntervalSamples) &&
+                    (m_InternalState != StateMachine::afterBeatActive) &&
+                    (m_InternalState != StateMachine::afterBeatDirectionChanged)) {
+                m_InternalState = StateMachine::afterBeatActive;
+                m_pCOBeatActive->forceSet(1.0);
+            } else if ((currentSample > prevIndicatorSamples +
+                                       m_blinkIntervalSamples) &&
+                    ((m_InternalState == StateMachine::afterBeatActive) ||
+                            (m_InternalState == StateMachine::afterBeatDirectionChanged))) {
+                m_InternalState = StateMachine::outsideIndicationArea;
+                m_pCOBeatActive->forceSet(0.0);
+            }
+        } else {
+            // Play direction changed while beat indicator was on and forward playing
+            if ((currentSample < nextIndicatorSamples) &&
+                    (currentSample >=
+                            nextIndicatorSamples - m_blinkIntervalSamples) &&
+                    (m_InternalState != StateMachine::beforeBeatDirectionChanged)) {
+                m_InternalState = StateMachine::beforeBeatDirectionChanged;
+                m_pCOBeatActive->forceSet(0.0);
+            }
+        }
+        m_lastPlayDirection = true; // Forward
+    } else if (dRate < 0.0) {
+        if (m_lastPlayDirection == false) {
+            if ((currentSample < nextIndicatorSamples) &&
+                    (currentSample >
+                            nextIndicatorSamples - m_blinkIntervalSamples) &&
+                    (m_InternalState != StateMachine::beforeBeatActive) &&
+                    (m_InternalState != StateMachine::beforeBeatDirectionChanged)) {
+                m_InternalState = StateMachine::beforeBeatActive;
+                m_pCOBeatActive->forceSet(-1.0);
+            } else if ((currentSample < nextIndicatorSamples -
+                                       m_blinkIntervalSamples) &&
+                    ((m_InternalState == StateMachine::beforeBeatActive) ||
+                            (m_InternalState == StateMachine::beforeBeatDirectionChanged))) {
+                m_InternalState = StateMachine::outsideIndicationArea;
+                m_pCOBeatActive->forceSet(0.0);
+            }
+        } else {
+            // Play direction changed while beat indicator was on and reverse playing
+            if ((currentSample > prevIndicatorSamples) &&
+                    (currentSample <=
+                            prevIndicatorSamples + m_blinkIntervalSamples) &&
+                    (m_InternalState != StateMachine::afterBeatDirectionChanged)) {
+                m_InternalState = StateMachine::afterBeatDirectionChanged;
+                m_pCOBeatActive->forceSet(0.0);
+            }
+        }
+        m_lastPlayDirection = false; // Reverse
     }
```

Comment:
I think I would prefer  
```suggestion
    } else {
        m_pCOBeatActive->forceSet(0.0);
    }
```
Because I experience that the LED is sometimes on and sometimes off when paused at a cue point.

Reply:
I will look how to solve the cue point issue. The proposed else branch will not do this. It will never be reached, because this is captured before, to not switch the indicator off due the jitter of the DVS pitch.

Reply:
I applied a fix for this, which should not break the use case of searching the beat position using DVS or jog. Please test, if this solves the cue point issue for you too.